### ISSUE TITLE
在 Manjaro 上添加`pybind11`后编译命令错误无法找到`stdlib.h`

### ISSUE THREAD

**Issue by myuanz** [xmake-io/xmake#4537]
### Xmake 版本

v2.8.5+20231107

### 操作系统版本和架构

Manjaro Linux 23.1.1

### 描述问题

使用 `xmake create` 创建一个初始项目后，写下：

```diff
add_rules("mode.debug", "mode.release")

+add_requires("pybind11")

target("test-xmake")
    set_kind("binary")
+   add_packages("pybind11")
    add_files("src/*.cpp")
```

在 Manjaro 下开始编译找不到`stdlib.h`。

在 Windows 10 和 `docker.io/library/ubuntu:latest` 下可正常编译

### 期待的结果

添加 pybind11 包后在 Manjaro 能正确找到 `stdlib.h`，编译命令不要错误包含`-isystem /usr/lib/pkgconfig/../../include`

### 工程配置

如上

### 附加信息和错误日志

添加 pybind11 时，编译日志为：

```
checking for gcc ... /usr/bin/gcc
checking for zig ... /usr/bin/zig
checking for unzip ... /usr/bin/unzip
checking for git ... /usr/bin/git
checking for gzip ... /usr/bin/gzip
checking for tar ... /usr/bin/tar
/usr/bin/git rev-parse HEAD
checking for cmake ... no
checking for cmake ... /usr/bin/cmake
finding libffi from xmake ..
checking for xmake::libffi ... no
finding libffi-dev from apt ..
checkinfo: cannot runv(dpkg --version), No such file or directory
checking for dpkg ... no
checking for apt::libffi-dev ... no
finding libffi from pacman ..
checking for pacman ... /usr/bin/pacman
checking for pkg-config ... /usr/bin/pkg-config
checking for pacman::libffi ... libffi 3.4.4
finding zlib from xmake ..
checking for xmake::zlib ... zlib v1.3
finding openssl from xmake ..
checking for xmake::openssl ... no
finding openssl from find_openssl ..
checkinfo: cannot runv(emerge --version), No such file or directory
checking for emerge ... no
finding openssl from brew ..
checking for brew ... no
finding openssl from vcpkg ..
finding openssl from conan ..
finding openssl from pkgconfig ..
checking for openssl ... pkgconfig::openssl 3.2.0
checking for xmake-repo::openssl ... openssl 3.2.0
finding ca-certificates from xmake ..
checking for xmake::ca-certificates ... no
finding ca-certificates from brew ..
finding ca-certificates from vcpkg ..
finding ca-certificates from conan ..
finding ca-certificates from pkgconfig ..
finding ca-certificates from pacman ..
checking for ca-certificates ... pacman::ca-certificates 20220905
finding python from xmake ..
checking for xmake::python ... no
checking for python3 ... /usr/bin/python3
checking for xmake-repo::python ... python 3.11.6
finding pybind11 from xmake ..
checking for xmake::pybind11 ... no
finding pybind11 from brew ..
finding pybind11 from vcpkg ..
finding pybind11 from conan ..
finding pybind11 from pkgconfig ..
checking for pybind11 ... pkgconfig::pybind11 2.11.1
checking for gcc ... /usr/bin/gcc
checking for the c++ compiler (cxx) ... gcc
checking for /usr/bin/gcc ... ok
checking for flags (-fPIC) ... ok
> gcc "-fPIC" "-m64"
checking for flags (-fvisibility-inlines-hidden) ... ok
> gcc "-fvisibility-inlines-hidden" "-m64"
checking for flags (-O3) ... ok
> gcc "-O3" "-m64"
checking for flags (-DNDEBUG) ... ok
> gcc "-DNDEBUG" "-m64"
[ 25%]: cache compiling.release src/main.cpp
/usr/bin/gcc -c -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -isystem /usr/lib/pkgconfig/../../include -isystem /usr/include/python3.11 -DNDEBUG -o build/.objs/test-xmake/linux/x86_64/release/src/main.cpp.o src/main.cpp
checking for flags (-MMD -MF) ... ok
> gcc "-MMD" "-MF" "/dev/null" "-m64"
checking for flags (-fdiagnostics-color=always) ... ok
> gcc "-fdiagnostics-color=always" "-m64"
error: @programdir/core/main.lua:309: @programdir/actions/build/main.lua:148: @programdir/modules/async/runjobs.lua:320: @programdir/modules/private/action/build/object.lua:91: @programdir/modules/core/tools/gcc.lua:841: In file included from /usr/include/c++/13.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/13.2.1/bits/basic_string.h:4097,
                 from /usr/include/c++/13.2.1/string:54,
                 from /usr/include/c++/13.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/13.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/13.2.1/ios:44,
                 from /usr/include/c++/13.2.1/ostream:40,
                 from /usr/include/c++/13.2.1/iostream:41,
                 from src/main.cpp:1:
/usr/include/c++/13.2.1/cstdlib:79:15: 致命错误：stdlib.h：没有那个文件或目录
   79 | #include_next <stdlib.h>
      |               ^~~~~~~~~~
编译中断。
stack traceback:
    [C]: in function 'error'
    [@programdir/core/base/os.lua:923]:
    [@programdir/modules/core/tools/gcc.lua:841]: in function 'catch'
    [@programdir/core/sandbox/modules/try.lua:123]: in function 'try'
    [@programdir/modules/core/tools/gcc.lua:782]:
    [C]: in function 'xpcall'
    [@programdir/core/base/utils.lua:280]:
    [@programdir/core/tool/compiler.lua:278]: in function 'compile'
    [@programdir/modules/private/action/build/object.lua:91]: in function 'script'
    [@programdir/modules/private/action/build/object.lua:122]: in function 'build_object'
    [@programdir/modules/private/action/build/object.lua:147]: in function 'jobfunc'
    [@programdir/modules/async/runjobs.lua:237]:
    [C]: in function 'xpcall'
    [@programdir/core/base/utils.lua:280]: in function 'trycall'
    [@programdir/core/sandbox/modules/try.lua:117]: in function 'try'
    [@programdir/modules/async/runjobs.lua:220]: in function 'cotask'
    [@programdir/core/base/scheduler.lua:404]:

stack traceback:
        [C]: in function 'error'
        @programdir/core/base/os.lua:923: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir/core/main.lua:309: in upvalue 'cotask'
        @programdir/core/base/scheduler.lua:404: in function <@programdir/core/base/scheduler.lua:397>
```
省去 pybind11 后，编译日志为：

```
checking for gcc ... /usr/bin/gcc
checking for zig ... /usr/bin/zig
checking for unzip ... /usr/bin/unzip
checking for git ... /usr/bin/git
checking for gzip ... /usr/bin/gzip
checking for tar ... /usr/bin/tar
/usr/bin/git rev-parse HEAD
checking for cmake ... no
checking for cmake ... /usr/bin/cmake
finding libffi from xmake ..
checking for xmake::libffi ... no
finding libffi-dev from apt ..
checkinfo: cannot runv(dpkg --version), No such file or directory
checking for dpkg ... no
checking for apt::libffi-dev ... no
finding libffi from pacman ..
checking for pacman ... /usr/bin/pacman
checking for pkg-config ... /usr/bin/pkg-config
checking for pacman::libffi ... libffi 3.4.4
finding zlib from xmake ..
checking for xmake::zlib ... zlib v1.3
finding openssl from xmake ..
checking for xmake::openssl ... no
finding openssl from find_openssl ..
checkinfo: cannot runv(emerge --version), No such file or directory
checking for emerge ... no
finding openssl from brew ..
checking for brew ... no
finding openssl from vcpkg ..
finding openssl from conan ..
finding openssl from pkgconfig ..
checking for openssl ... pkgconfig::openssl 3.2.0
checking for xmake-repo::openssl ... openssl 3.2.0
finding ca-certificates from xmake ..
checking for xmake::ca-certificates ... no
finding ca-certificates from brew ..
finding ca-certificates from vcpkg ..
finding ca-certificates from conan ..
finding ca-certificates from pkgconfig ..
finding ca-certificates from pacman ..
checking for ca-certificates ... pacman::ca-certificates 20220905
finding python from xmake ..
checking for xmake::python ... no
checking for python3 ... /usr/bin/python3
checking for xmake-repo::python ... python 3.11.6
finding pybind11 from xmake ..
checking for xmake::pybind11 ... no
finding pybind11 from brew ..
finding pybind11 from vcpkg ..
finding pybind11 from conan ..
finding pybind11 from pkgconfig ..
checking for pybind11 ... pkgconfig::pybind11 2.11.1
checking for gcc ... /usr/bin/gcc
checking for the c++ compiler (cxx) ... gcc
checking for /usr/bin/gcc ... ok
checking for flags (-fPIC) ... ok
> gcc "-fPIC" "-m64"
checking for flags (-fvisibility-inlines-hidden) ... ok
> gcc "-fvisibility-inlines-hidden" "-m64"
checking for flags (-O3) ... ok
> gcc "-O3" "-m64"
checking for flags (-DNDEBUG) ... ok
> gcc "-DNDEBUG" "-m64"
[ 25%]: cache compiling.release src/main.cpp
/usr/bin/gcc -c -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG -o build/.objs/test-xmake/linux/x86_64/release/src/main.cpp.o src/main.cpp
checking for flags (-MMD -MF) ... ok
> gcc "-MMD" "-MF" "/dev/null" "-m64"
checking for flags (-fdiagnostics-color=always) ... ok
> gcc "-fdiagnostics-color=always" "-m64"
checking for g++ ... /usr/bin/g++
checking for the linker (ld) ... g++
checking for /usr/bin/g++ ... ok
checking for flags (-fPIC) ... ok
> g++ "-fPIC" "-m64" "-m64"
[ 50%]: linking.release test-xmake
/usr/bin/g++ -o build/linux/x86_64/release/test-xmake build/.objs/test-xmake/linux/x86_64/release/src/main.cpp.o -m64 -s
```

在 Manjaro 中关键编译命令为
```bash
/usr/bin/gcc -c -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -isystem /usr/lib/pkgconfig/../../include -isystem /usr/include/python3.11 -DNDEBUG -o build/.objs/test-xmake/linux/x86_64/release/src/main.cpp.o src/main.cpp
```
在 Ubuntu 里是：
```bash
/usr/bin/gcc -c -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -isystem /root/.xmake/packages/p/pybind11/v2.10.0/41ef60b323db424f8cf3967e35d79f9e/include -isystem /root/.xmake/packages/p/python/3.11.3/d8c248d43f4141a9b452cf40280c9f68/include/python3.11 -isystem /root/.xmake/packages/o/openssl/1.1.1-w/6c51ab6278e2479b883dffafac69fdaf/include -DNDEBUG -o build/.objs/test-xmake/linux/x86_64/release/src/main.cpp.o src/main.cpp
```

注意到 Mnajaro 中多出了`-isystem /usr/lib/pkgconfig/../../include`，经尝试，删去后可正常编译

**Comment by waruqi** [comment]
里面依赖的 libffi python openssl 都被 pkg-config 从系统中找到，直接用了。。其中某个 pkg-config 吐了那个路径导致。。

要么走 `add_requireconfs("pybind11.**", {system = false, override = true})` 让依赖库全走 xmake-repo 仓库。要么找到哪个依赖，单将这个依赖 禁用系统库

**Comment by myuanz** [comment]
谢谢，经过尝试确实如此，我没有注意到那里实际使用了系统自带的 pybind11。

另外 override 这个参数不存在，有个警告`warning: add_requires("pybind11") has unknown option: {override=true}!`，我看了下文档应该是用 `add_requireconfs` 对吗？

**Comment by Freed-Wu** [comment]
I meet same error in ArchLinux. 

```
$ xmake --verbose
# ...
/usr/bin/gcc -c -m64 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -O3 -std=c++11 -Ibuild/.gens/example/linux/x86_64/release/platform/windows/idl -isystem /usr/lib/pkgconfig/../../include -isystem /usr/include/python3.13 -DNDEBUG -o build/.objs/example/linux/x86_64/release/example.cpp.o example.cpp
error: In file included from /usr/include/c++/14.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/14.2.1/bits/basic_string.h:4154,
                 from /usr/include/c++/14.2.1/string:54,
                 from /usr/include/c++/14.2.1/stdexcept:39,
                 from /usr/lib/pkgconfig/../../include/pybind11/detail/common.h:319,
                 from /usr/lib/pkgconfig/../../include/pybind11/attr.h:13,
                 from /usr/lib/pkgconfig/../../include/pybind11/detail/class.h:12,
                 from /usr/lib/pkgconfig/../../include/pybind11/pybind11.h:12,
                 from example.cpp:1:
/usr/include/c++/14.2.1/cstdlib:79:15: fatal error: stdlib.h: No such file or directory
   79 | #include_next <stdlib.h>
      |               ^~~~~~~~~~
compilation terminated.
$ pkg-config --cflags pybind11
-I/usr/lib/pkgconfig/../../include
$ pkg-config --cflags python3
-I/usr/include/python3.13
```

xmake replace `-I` to `-isystem` of the result of `pkg-config`, which make this error.
I doubt this behavior is incorrect. Change `-isystem` back to `-I` can make compilation pass.

**Comment by star-hengxing** [comment]
> I meet same error in ArchLinux.
> 
> ```
> $ xmake --verbose
> # ...
> /usr/bin/gcc -c -m64 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -O3 -std=c++11 -Ibuild/.gens/example/linux/x86_64/release/platform/windows/idl -isystem /usr/lib/pkgconfig/../../include -isystem /usr/include/python3.13 -DNDEBUG -o build/.objs/example/linux/x86_64/release/example.cpp.o example.cpp
> error: In file included from /usr/include/c++/14.2.1/ext/string_conversions.h:43,
>                  from /usr/include/c++/14.2.1/bits/basic_string.h:4154,
>                  from /usr/include/c++/14.2.1/string:54,
>                  from /usr/include/c++/14.2.1/stdexcept:39,
>                  from /usr/lib/pkgconfig/../../include/pybind11/detail/common.h:319,
>                  from /usr/lib/pkgconfig/../../include/pybind11/attr.h:13,
>                  from /usr/lib/pkgconfig/../../include/pybind11/detail/class.h:12,
>                  from /usr/lib/pkgconfig/../../include/pybind11/pybind11.h:12,
>                  from example.cpp:1:
> /usr/include/c++/14.2.1/cstdlib:79:15: fatal error: stdlib.h: No such file or directory
>    79 | #include_next <stdlib.h>
>       |               ^~~~~~~~~~
> compilation terminated.
> $ pkg-config --cflags pybind11
> -I/usr/lib/pkgconfig/../../include
> $ pkg-config --cflags python3
> -I/usr/include/python3.13
> ```
> 
> xmake replace `-I` to `-isystem` of the result of `pkg-config`, which make this error. I doubt this behavior is incorrect. Change `-isystem` back to `-I` can make compilation pass.

Try `--policies=package.include_external_headers:n`

**Comment by Freed-Wu** [comment]
```
xmake config --policies=package.include_external_headers:n
```

can stop `-I` to be changed to `-isystem`.
